### PR TITLE
chore: bump revisions

### PR DESCRIPTION
**Pull request by sakulstra**
This branch will be used for 3.0.1 upgrades on various networks

Personally i think this should be considered to be merged as not bumping revisions forces ppl to do weird tricks(like this one) or to copy contracts. To me it would seems reasonable to bump the revisions as for new deployments starting with `rev x` is not a problem, but for existing ones the static rev 0x1 makes upgrades harder.

Imo revision should reflect the revision of the contract. Not the number of deployments on a specific network.

It's `641c12a89556645fa8705d7d821060fcad1aea4b` with applied rev changes.

### CODE REVIEW

**Comment by miguelmtzinf** [comment]
This is a controversial topic. We have discussed about it multiple times and in my opinion there is no good approach here.

However, I believe that the less confusing approach is:
1. Keep the `main` branch with the last source code, with all revisions to 1. This is the code that is supposed to be deployed in case of a brand new pool.
2. Whenever a new deployment happens, create a tag with the corresponding used commit and document appropriately.
3. For upgrades, create a branch from `main` or the desired commit and apply the required changes to revisions. Create a tag for it.

TLDR: I prefer to merge these changes to a separate branch, that will be tagged as deployment.

**Comment by kartojal** [comment]
Im not very opiniated into this, but sharing what would be the dev flow in both cases:

- Cons of `REVISION` 1 at master branch
  - if we want to keep revision 1, them every time there is an implementation update the developer would need to:
    - Create the upgrade, with revision 1 and PR to `master`
    - Create another branch from the upgrade branch, and adjust the corresponding REVISION
 - Cons of `REVISION` *latest* at master branch
   - Maybe there is different revisions per market due special conditions could happen in specific market.
  
IMO revision usage should be deprecated, it just causes more pain to development and deployments and i really not sure why is there in first instance

**Comment by miguelmtzinf** [comment]
> IMO revision usage should be deprecated, it just causes more pain to development and deployments and i really not sure why is there in first instance

It is needed for upgrades. 

I would keep `REVISION 1` at master branch because:
- in this way you have at master the latest code deployed (which should be the same across networks) and specific tags for each deployment.
- if a new deployment happens, the master branch is up-to-date and ready with the correct code

**Comment by sakulstra** [comment]
Didn't see but also won't die on that hill as in the end it's annoying but not so problematic.
Essentially agree with @kartojal and disagree with the argument/preference of @miguelmtzinf (which is why i raised this issue in the first instance).

> It is needed for upgrades.

I mean it's a choice to need the revision, upgrades would work perfectly fine without it i guess.

The current behavior is very misleading.
- current master is not what is deployed anywhere (except ethereum v3, and even this is untrue when diffing as some docs seem to have changed)
- current master sais `REVISION 1` like every other pool implying they are the same - which they are not
- by forcing it 1, all pools eventually will have different revisions with similar, but not equal code

For me this `pro REVISION 1` argumentation is hard to understand and follow tbh.
You wouldn't argue for keeping any other (non solidity) code at version 1, because it's master.

> Revision: a revised edition or form of something.

The current `REVISION 1` is the revised edition of was previously `REVISION 1`, but with the mistake of not bumping the revision :shrug:
